src/components/Perspectives.tsx

```diff
@@ -1,14 +1,271 @@
+  const {state: {
+    client
+  }} = useContext(AgentContext);
+  
+  const [opened, handlers] = useDisclosure(false);
+
+  const deletePerspective = async (id: string) => {
+    await client!.perspective.remove(id);
+
+    await reload();
+
+    handlers.close();
+  }
+
+
+  return (
+    <Menu opened={opened} onOpen={handlers.open} onClose={handlers.close}>
+      <Menu.Item 
+        icon={<Trash size={16}/>}
+        onClick={() => deletePerspective(uuid)}
+      >
+        Delete
+      </Menu.Item>
+    </Menu>
+  )
+}
+
 const Perspectives = (props: Props) => {
+  const {state: {
+    client
+  }} = useContext(AgentContext);
+
+  const [perspectiveModalOpen, setPerspectiveModalOpen] = useState(false);
+  const [languages, setLanguages] = useState<LanguageHandle[] | null[]>([]);
+  const [perspectives, setPerspectives] = useState<PerspectiveProxy[] | null[]>([]);
+  const [perspectiveName, setPerspectiveName] = useState("");
+  const [isNeighbourhood, setIsNeighbourhood] = useState(false)
+  const [linkLanguage, setLinkLanguage] = useState('');
+  const [linkLanguages, setLinkLanguages] = useState<string[]>([]);
+  const [loading, setLoading] = useState(false);
+  
+  const fetchPerspective = async () => {
+    const perspectives = await client!.perspective.all();
+
+    console.log(perspectives)
+
+    setPerspectives(perspectives)
+  }
+
+  const getLanguages = async () => {
+    const langs = await client!.languages.all();
+
+    setLanguages(langs);
+  }
+  
+  useEffect(() => {
+    fetchPerspective()
+    getLanguages()
+  }, [])
+
+  const create = async () => {
+    setLoading(true)
+
+    const perspective = await client!.perspective.add(perspectiveName);
+    
+    try {
+      if (isNeighbourhood) {
+        const templateLangs = []
+  
+        const templatedLinkLang = await client!.languages.applyTemplateAndPublish(
+          linkLanguage,
+          JSON.stringify({
+            uid: '123',
+            name: `${perspectiveName}-social-context`
+          })
+        );
+  
+        const metaPerspective = await client!.perspective.add(`${perspectiveName}-meta`);
+
+        for (const linkLanguage of linkLanguages) {
+          const lang = (languages as LanguageHandle[]).find((e: LanguageHandle) => e.address === linkLanguage);
+  
+          if (lang) {
+  
+            const templatedLang = await client!.languages.applyTemplateAndPublish(
+              lang.address,
+              JSON.stringify({
+                uid: '123',
+                name: `${perspectiveName}-social-context`
+              })
+            );
+  
+            const link = await client!.perspective.addLink(metaPerspective.uuid, new Link({
+              source: 'self',
+              target: templatedLang.address,
+              predicate: 'language'
+            }));
+  
+            templateLangs.push(sanatizeLink(link))
+          }
+        }
+
+        await client!.perspective.remove(metaPerspective.uuid)
```

Comment:
Why remove the perspective here?

Reply:
this is a temporary perspective we use to publish links before we attach them to the neighbourhood.